server2/Arcadia.Assistant/Arcadia.Assistant.Sharepoint/Program.cs

```diff
@@ -0,0 +1,102 @@
+namespace Arcadia.Assistant.Sharepoint
+{
+    using System;
+    using System.Diagnostics;
+    using System.Fabric;
+    using System.Linq;
+    using System.Threading;
+
+    using Autofac;
+    using Autofac.Extensions.DependencyInjection;
+    using Autofac.Integration.ServiceFabric;
+
+    using Employees.Contracts;
+
+    using ExternalStorages.Abstractions;
+    using ExternalStorages.SharepointOnline;
+    using ExternalStorages.SharepointOnline.Contracts;
+
+    using Logging;
+
+    using Microsoft.Extensions.DependencyInjection;
+    using Microsoft.Extensions.Logging;
+    using Microsoft.ServiceFabric.Services.Remoting.Client;
+
+    using Models;
+
+    using Organization.Contracts;
+
+    using SickLeaves.Contracts;
+
+    using Vacations.Contracts;
+
+    using WorkHoursCredit.Contracts;
+
+    internal static class Program
+    {
+        /// <summary>
+        ///     This is the entry point of the service host process.
+        /// </summary>
+        private static void Main()
+        {
+            ILogger? logger = null;
+            try
+            {
+                var configurationPackage = FabricRuntime.GetActivationContext().GetConfigurationPackageObject("Config");
+                var sharepointSection = configurationPackage.Settings.Sections["Sharepoint"];
+                var calendarEventIdField = sharepointSection.Parameters.TryGetValue("CalendarEventIdField", out var val) ? val.Value : null;
```

Comment:
So you did this twice. Once in `SharepointOnlineConfiguration` class and the other is here. I suppose it is better to create all settings instances before registration block and use value from them.

Reply:
SharepointOnlineConfiguration doesnt parse calendarEventIdField parameter any more.

Reply:
Ok.